### PR TITLE
Support multiple model classes in @admin.register fixer

### PR DESCRIPTION
**Pull request by UnknownPlatypus**
Addresses point number 2 (Multiple model registration) of #190 

Following a review on the initial @admin.register fixer, I've also updated documentation links to `/en/stable/releases/` to remove yellow/red banners.

### CODE REVIEW

File: src/django_upgrade/fixers/admin_register.py [hunk 1]
```diff
@@ -108,39 +108,38 @@ def visit_Call(
...
+
+        elif (  # admin.site.register((MyModel1, MyModel2), MyCustomAdmin)
+            len(node.args) == 2
+            and isinstance((model_tuple := node.args[0]), ast.Tuple)
+            and all(isinstance(elt, ast.Name) for elt in model_tuple.elts)
+            and isinstance((admin_arg := node.args[1]), ast.Name)
+            and not node.keywords
+        ):
+            model_names = {
+                elt.id for elt in model_tuple.elts if isinstance(elt, ast.Name)
```
Comment (UnknownPlatypus) [inline src/django_upgrade/fixers/admin_register.py:134]: Mypy doesn't seems to narrow type correctly with `all(isinstance(elt, ast.Name) for elt in model_tuple.elts` so I had to do the explicit `isinstance(elt, ast.Name)` again. I don't know if you know any other way to workaround this ?

**Comment by adamchainz** [comment]
> Following a review on the initial @admin.register fixer, I've also updated documentation links to `/en/stable/releases/` to remove yellow/red banners.

I'd prefer not to do this for documentation versions that are still online. It's better to link to the Django 3.1 version of the 3.1 release notes, rather than the Django 4.1 version of the 3.1 release notes, to avoid confusion. I only suggested it for Django 1.7 because it's offline and you found you needed to link to the `dev` version`, but `stable` is better there since it doesn't have the yellow "future version" banner. Can you undo the changes here? 😅

**Comment by adamchainz** [comment]
Also in general please split PR's if they touch orthogonal areas of the code.

File: src/django_upgrade/fixers/admin_register.py [hunk 2]
```diff
@@ -108,39 +108,38 @@ def visit_Call(
...
+            and isinstance((model_arg := node.args[0]), ast.Name)
+            and len(node.keywords) == 1
+            and node.keywords[0].arg == "admin_class"
+            and isinstance((admin_arg := node.keywords[0].value), ast.Name)
+        ):
+            model_names = {model_arg.id}
+
+        elif (  # admin.site.register((MyModel1, MyModel2), MyCustomAdmin)
+            len(node.args) == 2
+            and isinstance((model_tuple := node.args[0]), ast.Tuple)
```
Comment (adamchainz) [inline src/django_upgrade/fixers/admin_register.py:128]: We should also support lists

```suggestion
            and isinstance((model_tuple := node.args[0]), (ast.Tuple, ast.List))
```

File: tests/fixers/test_admin_register.py [hunk 3]
```diff
@@ -470,6 +454,80 @@ class AuthorAdmin(CustomModelAdmin):
     )
 
 
+def test_multiple_model_multiline_registration():
```
Comment (adamchainz) [inline tests/fixers/test_admin_register.py:457]: let's add a duplicate of this test where the models are originally registered in the other order, to check the sorting behaviour

File: tests/fixers/test_admin_register.py [hunk 4]
```diff
@@ -470,6 +454,80 @@ class AuthorAdmin(CustomModelAdmin):
...
+        @admin.register(MyModel1, MyModel2)
+        class MyCustomAdmin:
+            pass
+
+        """,
+        settings=settings,
+    )
+
+
+def test_multiple_model_tuple_registration():
```
Comment (adamchainz) [inline tests/fixers/test_admin_register.py:482]: add a duplicate of this test using a list of model classes

File: tests/fixers/test_admin_register.py [hunk 5]
```diff
@@ -470,6 +454,80 @@ class AuthorAdmin(CustomModelAdmin):
...
+        from django.contrib import admin
+        from myapp.models import MyModel1, MyModel2
+
+        @admin.register(MyModel1, MyModel2, MyModel3)
+        class MyCustomAdmin:
+            pass
+
+        """,
+        settings=settings,
+    )
```
Comment (adamchainz) [inline tests/fixers/test_admin_register.py:528]: works nicely :)

**Review by adamchainz — COMMENTED** [review]
Please update the docs example to show multiple models being combined into a single `admin.register()` call.

File: tests/fixers/test_admin_register.py (hunk 5 above)
Comment (UnknownPlatypus) [reply tests/fixers/test_admin_register.py]: Yeah that's pretty cool :smiley:

File: src/django_upgrade/fixers/admin_register.py [hunk 6]
```diff
@@ -83,14 +83,14 @@ def update_class_def(
...
-    if len(model_names) == 1:
+    if model_names:
         if decorated:
             i = reverse_find(tokens, i, name=OP, src="@")
         j, indent = extract_indent(tokens, i)
         insert(
             tokens,
             j,
-            new_src=f"{indent}@admin.register({model_names.pop()})\n",
+            new_src=f"{indent}@admin.register({', '.join(sorted(model_names))})\n",
```
Comment (adamchainz) [inline src/django_upgrade/fixers/admin_register.py:93]: imo squeezing too much into the f-string, will separate

**Pull request merged**